File: Strivers-A2Z-DSA-Sheet/01.Arrays/1.Easy/javascript/08.Linear_search.js
/*
I don't think anyone needs it's solution. The idea is to traverse the array using loop and when the element
is equal to k return the same
*/

// CODE:-
function linearSearch(arr, n, k) {
    for (let i = 0; i < n; i++) {
        if (arr[i] === k)
            return i;
    }
    return -1;
}

// TIME COMPLEXITY = O(N)
// SPACE COMPLEXITY = O(1)

File: Strivers-A2Z-DSA-Sheet/01.Arrays/1.Easy/python/08.Linear_search.py
"""
I don't think anyone needs it's solution. The idea is to traverse the array using loop and when the element
is equal to k return the same
"""

# CODE:-
from typing import List

def linearSearch(arr: List[int], n: int, k: int) -> int:
    for i in range(n):
        if arr[i] == k:
            return i
    return -1

# TIME COMPLEXITY = O(N)
# SPACE COMPLEXITY = O(1)
